test/FunctionalTests/Service/Microsoft/OData/Service/Parsing/NodeToExpressionTranslator.cs

```diff
@@ -348,6 +348,28 @@ public override Expression Visit(SingleResourceCastNode node)
         }
 
         /// <summary>
+        /// Translates a <see cref="SinglePropertyCastNode"/> into a corresponding <see cref="Expression"/>.
+        /// </summary>
+        /// <param name="node">The node to translate.</param>
+        /// <returns>The translated expression.</returns>
+        public override Expression Visit(SinglePropertyCastNode node)
+        {
+            WebUtil.CheckArgumentNull(node, "node");
+
+            // Whenever we encounter the type segment, we need to only verify that the MPV is set to 4.0 or higher.
+            // There is no need to check for request DSV, request MaxDSV since there are no protocol changes in
+            // the payload for uri's with type identifier.
+            this.verifyProtocolVersion(ODataProtocolVersion.V4);

```

Comment:
Do we still need these Protocol Version checks? Is the code used by the 3.x stack?